### DISCUSSION TITLE
facing issue while applying migration

### DISCUSSION THREAD

**Discussion by Amanb1145** [ory/hydra#3327] [Q&A] [answered]
I am trying to setup Ory hydra on production on Azure VM using Ubuntu 20.04 but got some issues while following the documentation-
The url which is provided in documentation is not working anymore-
`wget https://github.com/ory/hydra/releases/download/v2.0.1/hydra_v2.0.1-linux_64bit.tar.gz`

so I used this one -

`wget https://github.com/ory/hydra/releases/download/v2.0.1/hydra_2.0.1-linux_64bit.tar.gz`

after doing this I downloaded config file using this url-

`wget https://raw.githubusercontent.com/ory/hydra/v2.0.1/contrib/quickstart/5-min/hydra.yml`

and then added the following lines in hydra.yml file-

`dsn: postgres://hydra:b0qw68gr3Q@127.0.0.1:5432/hydra?sslmode=disable&max_conns=20&max_idle_conns=4 urls: self: issuer: https://oauth2.example.com`

then I tried to apply the migration using-
`/opt/hydra/bin/hydra -c /opt/hydra/config/hydra.yml migrate sql -y postgres://hydra:b0qw68gr3Q@127.0.0.1:5432/hydra?sslmode=disable`

but getting the following error-

````
ERRO[2022-10-28T11:53:53Z] Unable to create service registry.            audience=application error=map[message:failed to connect to `host=127.0.0.1 user=hydra database=hydra`: failed SASL auth (FATAL: password authentication failed for user "hydra" (SQLSTATE 28P01))] service_name=Ory Hydra service_version=v2.0.1
Error: failed to connect to `host=127.0.0.1 user=hydra database=hydra`: failed SASL auth (FATAL: password authentication failed for user "hydra" (SQLSTATE 28P01))
Usage:
  hydra migrate sql <database-url> [flags]

Flags:
  -h, --help            help for sql
  -e, --read-from-env   If set, reads the database connection string from the environment variable DSN or config file key dsn.
  -y, --yes             If set all confirmation requests are accepted without user interaction.

Global Flags:
  -c, --config strings   Path to one or more .json, .yaml, .yml, .toml config files. Values are loaded in the order provided, meaning that the last config file overwrites values from the previous config file.

failed to connect to `host=127.0.0.1 user=hydra database=hydra`: failed SASL auth (FATAL: password authentication failed for user "hydra" (SQLSTATE 28P01))
```
`

Thank You

**Comment by vinckr** [comment, marked as answer]
Hello @Amanb1145 
Apologies for the late response, how is it going for you now? 

Based on the error message you provided, it seems like the issue is related to the authentication with your PostgreSQL database. 
The error message FATAL: password authentication failed for user "hydra" indicates that the password provided for the user "hydra" is incorrect or the user does not exist in your PostgreSQL database.

Here are some steps you can follow to troubleshoot this issue:
- Check if the PostgreSQL user "hydra" exists. You can do this by logging into your PostgreSQL database and running the following command: \du. This will list all users in your PostgreSQL database.
- If the user "hydra" does not exist, you need to create it. You can do this by running the following command in your PostgreSQL database: `CREATE USER hydra PASSWORD '<YOUR_PASSWORD_HERE>';` [source](https://www.ory.sh/docs/hydra/self-hosted/deploy-hydra-example)
- If the user "hydra" exists, check if the password provided in your dsn string in the hydra.yml file is correct. The password should be the same as the one you used when creating the "hydra" user.
- Also, ensure that you have created a database named "hydra" in your PostgreSQL database. You can do this by running the following command in your PostgreSQL database: `createdb hydra` 
- After ensuring that the user exists and the password is correct, try running the migration command again.
- If you continue to face issues, you can check the logs for more detailed error messages. Ory Hydra has extensive logging and you will most likely find the issue in the logs [source](https://www.ory.sh/docs/hydra/faq).

Also, ensure that you have followed all the steps in the Ory Hydra documentation for setting up the PostgreSQL database and Ory Hydra correctly.
